src/coreclr/inc/clrnt.h

```diff
@@ -1028,6 +1028,71 @@ RtlVirtualUnwind(
 #define IMAGE_REL_ARM64_PAGEOFFSET_12A  0x0006  // ADD 12 bit page offset
 #endif
 
+#endif // TARGET_ARM64
+
+#ifdef TARGET_LOONGARCH64
+#include "daccess.h"
+
+#define UNW_FLAG_NHANDLER               0x0             /* any handler */
+#define UNW_FLAG_EHANDLER               0x1             /* filter handler */
+#define UNW_FLAG_UHANDLER               0x2             /* unwind handler */
+
+// This function returns the RVA of the end of the function (exclusive, so one byte after the actual end)
+// using the unwind info on LOONGARCH64. (see ExternalAPIs\Win9CoreSystem\inc\winnt.h)
+FORCEINLINE
+ULONG64
+RtlpGetFunctionEndAddress (
+    __in PT_RUNTIME_FUNCTION FunctionEntry,
+    __in ULONG64 ImageBase
+    )
+{
+    ULONG64 FunctionLength;
+
+    FunctionLength = FunctionEntry->UnwindData;
+    if ((FunctionLength & 3) != 0) {
+        FunctionLength = (FunctionLength >> 2) & 0x7ff;
+    } else {
+        memcpy(&FunctionLength, (void*)(ImageBase + FunctionLength), sizeof(UINT32));
+        FunctionLength &= 0x3ffff;
+    }
+
+    return FunctionEntry->BeginAddress + 4 * FunctionLength;
+}
+
+#define RUNTIME_FUNCTION__BeginAddress(FunctionEntry)               ((FunctionEntry)->BeginAddress)
+#define RUNTIME_FUNCTION__SetBeginAddress(FunctionEntry,address)    ((FunctionEntry)->BeginAddress = (address))
+
+#define RUNTIME_FUNCTION__EndAddress(FunctionEntry, ImageBase)      (RtlpGetFunctionEndAddress(FunctionEntry, (ULONG64)(ImageBase)))
+
+#define RUNTIME_FUNCTION__SetUnwindInfoAddress(prf,address)         do { (prf)->UnwindData = (address); } while (0)
+
+typedef struct _UNWIND_INFO {
+    // dummy
+} UNWIND_INFO, *PUNWIND_INFO;
+
+EXTERN_C
+NTSYSAPI
+PEXCEPTION_ROUTINE
+NTAPI
+RtlVirtualUnwind(
+    IN ULONG HandlerType,
+    IN ULONG64 ImageBase,
+    IN ULONG64 ControlPc,
+    IN PRUNTIME_FUNCTION FunctionEntry,
+    IN OUT PCONTEXT ContextRecord,
+    OUT PVOID *HandlerData,
+    OUT PULONG64 EstablisherFrame,
+    IN OUT PKNONVOLATILE_CONTEXT_POINTERS ContextPointers OPTIONAL
+    );
+
+#ifndef IMAGE_FILE_MACHINE_LOONGARCH64
```

Comment:
I had add `IMAGE_FILE_MACHINE_LOONGARCH64 ` here, but compiling error is also exist.

Reply:
You should add it outside the `#ifdef TARGET_LOONGARCH64 / #endif` block

Reply:
Thanks.
Now I had broken up this PR into separate PRs.
So this file will be moved to vm directory because of the ToolBox directory which it is related with jitinterface. #62885

Reply:
> You should add it outside the `#ifdef TARGET_LOONGARCH64 / #endif` block

But the `IMAGE_FILE_MACHINE_ARM64` is within the  `#ifdef TARGET_ARM64 / #endif` block.
Is the ARM64 supporting for windows ?

Reply:
Yes, we get `IMAGE_FILE_MACHINE_ARM64` definition from the Windows SDK headers, but it is not the case for LoongArch64.

Note that it would be fine for `IMAGE_FILE_MACHINE_ARM64` and other ARM64 definition to be outside the `#ifdef TARGET_ARM64
` / `#endif` too since it is protected against multiple definitions. If the definition comes from Windows SDK headers, it is skipped here. The definition in this file only kicks in if it does not come from Windows SDK or other headers.

Reply:
Thanks for your help~